conversations.go

```diff
@@ -0,0 +1,205 @@
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+
+	conversation, err := server.repo.Get(ctx, &models.OneConversationFilter{ID: in.ConversationId})
+	if err != nil {
+		return nil, statusFromModelError(err)
+	}
+
+	accountId := token.UserID.String()
+
+	_, err = server.groupService.GetGroupMember(ctx, &accountsv1.GetGroupMemberRequest{GroupId: conversation.GroupID, AccountId: accountId})
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, "sender not in group")
+	}
+
+	err = server.repo.Delete(ctx, &models.OneConversationFilter{ID: in.ConversationId})
+	if err != nil {
+		return nil, statusFromModelError(err)
+	}
+
+	return &accountsv1.DeleteConversationResponse{}, nil
+}
+
+func (server *conversationsAPI) UpdateConversation(ctx context.Context, in *accountsv1.UpdateConversationRequest) (*accountsv1.UpdateConversationResponse, error) {
+	token, err := server.authenticate(ctx)
+	if err != nil {
+		return nil, status.Error(codes.Unauthenticated, err.Error())
+	}
+
+	conversation, err := server.repo.Get(ctx, &models.OneConversationFilter{ID: in.ConversationId})
+	if err != nil {
+		return nil, statusFromModelError(err)
+	}
+
+	accountId := token.UserID.String()
+
+	_, err = server.groupService.GetGroupMember(ctx, &accountsv1.GetGroupMemberRequest{GroupId: conversation.GroupID, AccountId: accountId})
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, "sender not in group")
+	}
+
+	err = validators.ValidateUpdateConversationRequest(in)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+
+	filter := models.OneConversationFilter{ID: in.ConversationId}
+
+	updatedConversation, err := server.repo.Update(ctx, &filter, &models.UpdateConversationPayload{Title: in.Title})
+	if err != nil {
+		return nil, statusFromModelError(err)
+	}
+
+	return &accountsv1.UpdateConversationResponse{
+		Conversation: &accountsv1.Conversation{
+			Id:      in.ConversationId,
+			GroupId: updatedConversation.GroupID,
+			Title:   in.Title,
+		},
+	}, nil
+}
+
+func (server *conversationsAPI) ListConversations(ctx context.Context, in *accountsv1.ListConversationsRequest) (*accountsv1.ListConversationsResponse, error) {
+	token, err := server.authenticate(ctx)
+	if err != nil {
+		return nil, status.Error(codes.Unauthenticated, err.Error())
+	}
+
+	accountId := token.UserID.String()
+
+	_, err = server.groupService.GetGroupMember(ctx, &accountsv1.GetGroupMemberRequest{GroupId: in.GroupId, AccountId: accountId})
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, "sender not in group")
+	}
+
+	err = validators.ValidateListConversationRequest(in)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+
+	conversations, err := server.repo.List(ctx, &models.ManyConversationsFilter{GroupID: in.GroupId})
+	if err != nil {
+		return nil, statusFromModelError(err)
+	}
+
+	var groupConversations []*accountsv1.Conversation
```

Comment:
Difference between `nil` and `[]`. Front gets `{}` wants `{conversations: []}`.
```suggestion
	groupConversations := make([]*accountsv1.Conversation, 0)
```